src/v/raft/consensus.cc

```diff
@@ -3114,7 +3114,9 @@ void consensus::update_follower_stats(const group_configuration& cfg) {
 }
 
 void consensus::trigger_leadership_notification() {
-    _probe->leadership_changed();
+    if (_leader_id) {
```

Comment:
> This way after each successful leader election the metric is incremented by one on all partition replicas.

This cuts it down by a factor of 2 but is still accounted by all replicas (3x), I was wondering if we can do something like, 

if (_leader_id && _leader_id == _self) {
  metric++
}

basically account for successful leader transfers to that node.

Reply:
i think we should report the change from every replica as this is what the replica is experiencing

Reply:
i mean the follower sees the leadership change as well it is not limited to leader